timelock-server/src/main/java/com/palantir/atlasdb/timelock/clock/ClockServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.clock;
+
+public class ClockServiceImpl implements ClockService {
+    @Override
+    public long getSystemTimeInNanos() {
+        return System.nanoTime();
```

Comment:
For context: http://btorpey.github.io/blog/2014/02/18/clock-sources-in-linux/

I'd argue that `System.html#nanoTime()` seems to be better for this case. Note that we're doing exactly what the documentation states it's intended for, which is to measure how much time is spent between two calls to this method. Also, we probably want to trigger the underlying `CLOCK_MONOTONIC`, which does not use stepping when measuring time (abrupt changes).

In fact, we probably want neither of those methods. We probably want a method that also doesn't support slewing (changing the pace), which both of them seem to do... I'll consider add a JNI method to use the underlying `CLOCK_MONOTONIC_RAW`, which seems to be what we want...

Note also that this experiment (and the following optimization) should only be enabled in supported systems (systems that use the underlying CLOCK_MONOTONIC option).